main/options.c

```diff
@@ -2363,6 +2366,22 @@ static void processMaxRecursionDepthOption (const char *const option, const char
 	Option.maxRecursionDepth = atol(parameter);
 }
 
+static void processPatternLengthLimit(const char *const option, const char *const parameter)
+{
+	long patternLengthLimit;
+
+	if (parameter == NULL || parameter[0] == '\0')
+		error (FATAL, "A parameter is needed after \"%s\" option", option);
+
+	errno = 0;
+	patternLengthLimit = strtol(parameter, NULL, 0);
+
+	if (errno != 0 || patternLengthLimit < 0)
+		error (FATAL, "-%s: Invalid pattern length limit", option);
```

Comment:
@masatake @b4n I've added the requested changes.

@b4n If you like to have more acknowledgement in df7365d I'm happy to adapt the commit message.